bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductTeaserImpl.java

```diff
@@ -81,99 +69,79 @@ protected void initModel() {
         }
         String selection = properties.get(SELECTION_PROPERTY, String.class);
         if (selection != null && !selection.isEmpty()) {
+            if (selection.startsWith("/")) {
+                selection = StringUtils.substringAfterLast(selection, "/");
+            }
+            combinedSku = SiteNavigation.toProductSkus(selection);
+
             // Get MagentoGraphqlClient from the resource.
-            magentoGraphqlClient = MagentoGraphqlClient.create(resource);
+            MagentoGraphqlClient magentoGraphqlClient = MagentoGraphqlClient.create(resource);
 
             // Fetch product data
             if (magentoGraphqlClient != null) {
-                product = fetchProduct(selection);
+                productRetriever = new ProductRetriever(magentoGraphqlClient);
+                productRetriever.setIdentifier(combinedSku.getLeft());
             }
+        }
+    }
 
-            Locale locale = currentPage.getLanguage(false);
-            priceFormatter = Utils.buildPriceFormatter(locale, getCurrency());
+    private ProductInterface getProduct() {
+        ProductInterface baseProduct = productRetriever.getProduct();
+        if (combinedSku.getRight() != null && baseProduct instanceof ConfigurableProduct) {
+            ConfigurableProduct configurableProduct = (ConfigurableProduct) baseProduct;
+            SimpleProduct variant = findVariant(configurableProduct, combinedSku.getRight());
+            if (variant != null) {
+                return variant;
+            }
         }
+        return baseProduct;
     }
 
     @Override
     public String getName() {
-        return product != null ? product.getName() : null;
+        return getProduct().getName();
     }
 
     @Override
     public String getFormattedPrice() {
         Double price = getPrice();
         if (price != null) {
-            return priceFormatter.format(price);
+            return getPriceFormatter().format(price);
         }
         return null;
     }
 
     @Override
     public String getUrl() {
-        if (product != null) {
-            return SiteNavigation.toProductUrl(productPage.getPath(), baseProductSlug, variantSku);
+        if (getProduct() != null) {
+            return SiteNavigation.toProductUrl(productPage.getPath(), productRetriever.getProduct().getUrlKey(), combinedSku.getRight());
```

Comment:
Why don't you use `getProduct()` here like in other methods?

Reply:
The previous implementation was explicitly using the base product to get the slug. `getProduct()` could return a variant if the combined sku includes a variant sku.